client/app/views/partials/preferences/pgpkey.html

```diff
@@ -1,30 +1,31 @@
-<div data-ng-show="preferences.pgp_key_status === 'enabled'">
-  <div class="alert alert-success">
-    <span data-translate>Encryption is enabled!</span>
-    <span data-translate>Email notifications will be encrypted.</span>
-  </div>
+<div data-ng-form="pgpkeyForm">
+  <div data-ng-show="preferences.pgp_key_status === 'enabled'">
+    <div class="alert alert-success">
+      <span data-translate>Encryption is enabled!</span>
+      <span data-translate>Email notifications will be encrypted.</span>
+    </div>
 
-  <pgp-pubkey-display key-str="preferences.pgp_key_public"></pgp-pubkey-display>
-  <div class="form-group">
-    <input data-ng-model="preferences.pgp_key_remove" type="checkbox">
-    <label data-translate>Remove the PGP key</label>
+    <div pgp-pubkey-display key-str="preferences.pgp_key_public"></div>
+    <div class="form-group">
+      <input data-ng-model="preferences.pgp_key_remove" type="checkbox">
+      <label data-translate>Remove the PGP key</label>
+    </div>
   </div>
-</div>
 
-<div class="form-group" data-ng-hide="preferences.pgp_key_status === 'enabled'">
-  <div class="alert alert-danger">
-    <div data-translate>Encryption has not been enabled!</div>
-    <div data-translate>Email notifications won't be encrypted.</div>
+  <div class="form-group" data-ng-hide="preferences.pgp_key_status === 'enabled'" data-ng-class="{'has-error': !pgpkeyForm.$valid}">
```

Comment:
added the classes for showing the red borders in of missing required / invalid key.